### PR TITLE
MetalLB fetch branch automatically

### PR DESCRIPTION
**Pull request by liornoy**
This PR modifies the metallb-e2e-test-commands script
to fetch the branch out from the OC version.

Also, remove the METALLB_BRANCH and METALLB_OPERATOR_BRANCH
env vars from the yamls, as we don't need them anymore.

Note: this change is aiming to simplify the e2e CI and save
us from maintaining those env vars and updating them to the
correct "release" branch manually.

### CODE REVIEW

**Comment by liornoy** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-telco5g-cnftests

**Comment by liornoy** [comment]
/pj-rehearse pull-ci-openshift-metallb-release-4.14-metallb-e2e-metal-ipi

**Comment by liornoy** [comment]
/pj-rehearse pull-ci-openshift-metallb-operator-release-4.14-metallb-e2e-metal-ipi

**Comment by liornoy** [comment]
/pj-rehearse pull-ci-openshift-metallb-operator-release-4.14-metallb-e2e-metal-ipi

**Comment by liornoy** [comment]
the metallb operator failed due to certification error. retrying.

**Comment by liornoy** [comment]
/pj-rehearse pull-ci-openshift-metallb-operator-release-4.14-metallb-e2e-metal-ipi

**Comment by liornoy** [comment]
/pj-rehearse pull-ci-openshift-metallb-operator-release-4.14-metallb-e2e-metal-ipi

**Comment by liornoy** [comment]
/pj-rehearse pull-ci-openshift-metallb-release-4.14-metallb-e2e-metal-ipi

**Comment by liornoy** [comment]
/pj-rehearse pull-ci-openshift-metallb-operator-release-4.14-metallb-e2e-metal-ipi

**Comment by liornoy** [comment]
/pj-rehearse pull-ci-openshift-metallb-release-4.14-metallb-e2e-metal-ipi

**Comment by liornoy** [comment]
/pj-rehearse pull-ci-openshift-metallb-main-metallb-e2e-metal-ipi

**Comment by liornoy** [comment]
/pj-rehearse pull-ci-openshift-metallb-operator-main-metallb-e2e-metal-ipi

**Comment by liornoy** [comment]
/pj-rehearse pull-ci-openshift-frr-master-metallb-e2e-metal-ipi

**Comment by fedepaol** [comment]
@liornoy just to be on the safe side, can you reharse also a job on the operator repo and one from the frr one?

**Comment by liornoy** [comment]
> @liornoy just to be on the safe side, can you reharse also a job on the operator repo and one from the frr one?

Can you please clarify how do I rehearse a job with those changes on the operator and the frr repos?

**Comment by fedepaol** [comment]
> > @liornoy just to be on the safe side, can you reharse also a job on the operator repo and one from the frr one?
> 
> Can you please clarify how do I rehearse a job with those changes on the operator and the frr repos?

Here https://github.com/openshift/release/pull/41419#issuecomment-1646789294 the bot tells you what are the jobs affected and how to reharse them.
Just pick the ovn one for the frr and the operator main repos

**Comment by liornoy** [comment]
/pj-rehearse pull-ci-openshift-metallb-operator-main-metallb-e2e-metal-ipi-ovn

**Comment by liornoy** [comment]
/pj-rehearse pull-ci-openshift-frr-master-metallb-e2e-metal-ipi-ovn

**Comment by liornoy** [comment]
@fedepaol the tests passed.

**And to sum up what is done in this PR:**
We keep using the **METALLB_BRANCH** and **METALLB_OPERATOR_BRANCH** vars, but
feed it from the CI script (baremetalds-metallb-e2e-test-commands.sh) as opposed to assigning it 
values via the `ci-operator/config/...yaml`.
For that reason, this PR also removes the assignment of those vars in the related main/master YAMLs.

The only risk we're taking here is relying on the `oc version` output format that could change in the future
and break the CI, requiring us to fix the `extract_version()` bash function responsible for extracting the
branch name we need to assign.

File: ci-operator/config/openshift-priv/frr/openshift-priv-frr-master.yaml [hunk 1]
```diff
@@ -48,8 +48,6 @@ tests:
         NETWORK_TYPE=OpenShiftSDN
       E2E_TESTS_CONFIG: |
         IP_STACK=v4
-        METALLB_BRANCH=main
```
Comment (fedepaol) [inline ci-operator/config/openshift-priv/frr/openshift-priv-frr-master.yaml:51]: these are not changed by us. We normally change only ci-operator/config and run make jobs.

**Comment by fedepaol** [comment]
From the logs I see 

+(./deploy_metallb.sh:6): main(): METALLB_OPERATOR_REPO=https://github.com/openshift/metallb-operator.git
+(./deploy_metallb.sh:7): main(): METALLB_OPERATOR_BRANCH=main

This means that the changes in the script were not applied to this run (Which is totally legit, but also means that we'll have to merge this to see the effects)>

File: ci-operator/config/openshift-priv/frr/openshift-priv-frr-master.yaml (hunk 1 above)
Comment (fedepaol) [reply ci-operator/config/openshift-priv/frr/openshift-priv-frr-master.yaml:51]: I mean, we normally touch only config/openshift/ and not openshift-priv/

Comment (liornoy) [reply ci-operator/config/openshift-priv/frr/openshift-priv-frr-master.yaml:51]: oh ok. what is it for anyway? I'll fix it together with the commit title.

Comment (fedepaol) [reply ci-operator/config/openshift-priv/frr/openshift-priv-frr-master.yaml:51]: and https://github.com/openshift/release/commit/7e5ad3a10b5fea19a407268ef4ff3101f2f99f66

**Comment by liornoy** [comment]
/pj-rehearse pull-ci-openshift-frr-master-metallb-e2e-metal-ipi-ovn
/pj-rehearse pull-ci-openshift-metallb-operator-main-metallb-e2e-metal-ipi-ovn
/pj-rehearse pull-ci-openshift-metallb-main-metallb-e2e-metal-ipi-ovn
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-telco5g-cnftests

**Comment by liornoy** [comment]
/pj-rehearse pull-ci-openshift-metallb-operator-main-metallb-e2e-metal-ipi-ovn

**Comment by liornoy** [comment]
/pj-rehearse pull-ci-openshift-metallb-main-metallb-e2e-metal-ipi-ovn

**Comment by liornoy** [comment]
> From the logs I see
> 
> +(./deploy_metallb.sh:6): main(): METALLB_OPERATOR_REPO=https://github.com/openshift/metallb-operator.git +(./deploy_metallb.sh:7): main(): METALLB_OPERATOR_BRANCH=main
> 
> This means that the changes in the script were not applied to this run (Which is totally legit, but also means that we'll have to merge this to see the effects)>

Instead of the "export METALLB_OPERATOR_BRANCH" line, I added it to the args we pass to the ./deploy_metallb script.
Let's see now that it's valid.

**Comment by liornoy** [comment]
@fedepaol 
+(./deploy_metallb.sh:6): main(): METALLB_OPERATOR_REPO=https://github.com/openshift/metallb-operator.git
+(./deploy_metallb.sh:7): main(): METALLB_OPERATOR_BRANCH=release-4.14

I think we're good now.

**Pull request merged**